### ISSUE TITLE
Unable to connect to WSGI daemon process 'wsgi'

### ISSUE THREAD

**Issue by rahularorayourfrn** [GrahamDumpleton/mod_wsgi#697]
Hello there,

It seems that every time segmentation fault occurs in one of the third party libraries that wsgi(python) code is using, apache reloads the daemon process. However, the apache child process do not die ( may be because there are still holding active connection). 

This causes some of the requests to fail with error like :  "No such file or directory: [client [my ip]:XXX] mod_wsgi (pid=9734): Unable to connect to WSGI daemon process '[daemon process name]' on '/var/run/apache2/wsgi.2629.0.1.sock' as user with uid=33

The only way then is to restart the apache manually. 

Is there any way I can force the apache restart whenever apache reloads daemon process for any reason ?

Thank you

**Comment by GrahamDumpleton** [comment]
What you are describing could only happen if you are using a client which attempts to keep a permanent connection to the Apache process using keep alive connections and you have also set a large maximum for keep alive requests. Plus, you do an Apache graceful restart. The latter can be triggered by system logrotate service so you may not have done it yourself.

There is no way this could occur purely by a daemon process crashing as you describe.

Anyway, set:

```
WSGISocketRotation Off
```

See:

* https://modwsgi.readthedocs.io/en/master/release-notes/version-4.6.0.html?highlight=WSGISocketRotation#new-features

This will cause the same path for the socket to always be used even across Apache restarts, thus the still running Apache child worker processes should be able to connect to new iteration of mod_wsgi daemon processes.

**Comment by rahularorayourfrn** [comment]
Dear Graham, 

Thank you so much for your time and response. I spent some more time in between and found that actually log rotation is gracefully restarting the apache. 

Thanks for the WSGISocketRotation tip.

**Comment by chenz** [comment]
I also ran into this problem, with an out-of-the-box Apache + mod_wsgi configuration on Ubuntu. My understanding is also that keep-alive is the default these days. So this might be a more common problem than you might think.

WSGISocketRotation is missing in the configuration documentation, and is only mentioned in the release notes. I only found this option (and then this issue by web-searching for it) after looking through the sources to see what might cause the socket path to change.
